src/Libraries/RevitNodes/Application/FamilyDocument.cs

```diff
@@ -0,0 +1,283 @@
+
+        /// <summary>
+        /// Sets the Category object that represents the category or sub category in which the elements ( this family could generate ) reside.
+        /// </summary>
+        /// <param name="category"></param>
+        /// <returns>The document family</returns>
+        public FamilyDocument SetCategory(Category category)
+        {
+            TransactionManager.Instance.EnsureInTransaction(DocumentManager.Instance.CurrentDBDocument);
+            OwnerFamily.InternalFamily.FamilyCategory = category.InternalCategory;
+            TransactionManager.Instance.TransactionTaskDone();
+            return this;
+        }
+
+        /// <summary>
+        /// Set the formula of a family parameter.
+        /// </summary>
+        /// <param name="parameter">The family parameter.</param>
+        /// <param name="formula">The formula string.</param>
+        /// <returns>The document family</returns>
+        public FamilyDocument SetFormula(string parameter, string formula)
+        {
+            Autodesk.Revit.DB.FamilyParameter familyParameter = FamilyManager.get_Parameter(parameter);
+            if (familyParameter == null)
+                throw new InvalidOperationException(Properties.Resources.ParameterNotFound);
+
+            TransactionManager.Instance.EnsureInTransaction(DocumentManager.Instance.CurrentDBDocument);
+            FamilyManager.SetFormula(familyParameter, formula);
+            TransactionManager.Instance.TransactionTaskDone();
+            return this;
+        }
+
+        /// <summary>
+        /// Gets the formula of a family parameter.
+        /// </summary>
+        /// <param name="parameter">The family parameter.</param>
+        /// <returns>The family parameter formula.</returns>
+        public string GetFormula(string parameter)
+        {
+            Autodesk.Revit.DB.FamilyParameter familyParameter = FamilyManager.get_Parameter(parameter);
+            if (familyParameter == null)
+                throw new InvalidOperationException(Properties.Resources.ParameterNotFound);
+
+            return familyParameter.Formula;
+        }
+
+        /// <summary>
+        /// Get's the value of a family parameter of the current family type.
+        /// </summary>
+        /// <param name="parameter">A family parameter of the current type.</param>
+        /// <returns>The parameter value.</returns>
+        public object ParameterValueByName(string parameter)
+        {
+            Autodesk.Revit.DB.FamilyParameter familyParameter = FamilyManager.get_Parameter(parameter);
+            if (familyParameter == null)
+                throw new InvalidOperationException(Properties.Resources.ParameterNotFound);
+            switch (familyParameter.StorageType)
+            {
+                case Autodesk.Revit.DB.StorageType.Integer:
+                    return FamilyManager.CurrentType.AsInteger(familyParameter) * UnitConverter.HostToDynamoFactor(familyParameter.Definition.UnitType);   
```

Comment:
@radumg 
not sure i follow. It wont go through all cases but go directly to the right one, where it will get the currenType from the familyManager only ones, or am i misunderstanding something?

If i cache the FamilyManager.CurrentType before the switch it will always do this work even if it goes to the default case where the CurrentType is not needed.

Reply:
yep, that's correct, I was referring to C# issues not code logic, so was more referring to the fact that `FamilyManager` itself is a computed property, so every time you access it (or one of its props), it'll go up the chain to the class it inherits from, grab the other prop and then return `FamilyManager`.

A bit of a micro-optimisation i guess 🤷‍♂